### PR TITLE
[serverless] Add build option

### PR DESCRIPTION
**Pull request by tri-star**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
  - Document: https://www.serverless.com/framework/docs/providers/aws/guide/building#configuration
  - serverless CLI source code: https://github.com/serverless/serverless/blob/73b3de8a6aeeedf942c1c6fc2314417c82f149b7/lib/plugins/esbuild/index.js#L76-L115
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/serverless/plugins/esbuild/index.d.ts
1. types/serverless/serverless-tests.ts
1. types/serverless/plugins/aws/provider/awsProvider.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c7725a1c70bba883623bec65a533103fc6ccaeb5
</p>

File: types/serverless/package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
     "private": true,
     "name": "@types/serverless",
-    "version": "3.12.9999",
+    "version": "3.13.9999",
```
Comment (G-Rath) [inline types/serverless/package.json:4]: You don't need to do this as versioning is handled by the release bot - I don't think there's any harm, but I'd prefer not doing it to be on the safe side

**Review by G-Rath — CHANGES_REQUESTED** [review]
these types are for serverless v3, but this change is only in serverless v4 - we should do a dedicated pr updating to v4 before this can land

File: types/serverless/package.json (hunk 1 above)
Comment (tri-star) [reply types/serverless/package.json]: Thank you for your comment. I was not aware that the version number is automatically managed by the bot.
I have fixed it.

**Comment by tri-star** [comment]
@G-Rath
Thank you for your checking.
This time, I created a PR because I needed the build.esbuild option for the task I am in charge of. However, since I am not fully aware of the changes affecting the entire v4, I would like to withdraw this PR and contribute on another occasion.

If there are no issues, please allow me to close this.

**Comment by G-Rath** [comment]
@tri-star please close this per our discussion

**Comment by tri-star** [comment]
@G-Rath 
Sorry for the delay. I will close this PR.

If there is another opportunity to contribute in the future, please allow me to submit a PR again.

**Pull request closed without merging**